src/main/java/org/moditect/jfrunit/JfrAnalysis.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2020 - 2021 The JfrUnit authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.moditect.jfrunit;
```

Comment:
Is this class user-facing? Or should it be moved to the internal package?

Reply:
I am thinking about this now. Atm it is not user facing, but it is tightly coupled to `JfrEvents`.  I think it depends on the workflow you would expect users to generate the analysis.
i.e.
```
        jfrEvents.awaitEvents();

        List<IResult> analysisResults = JfrAnalysis.analyse(jfrEvents);

```
or
```
        jfrEvents.awaitEvents();

        List<IResult> analysisResults = jfrEvents.automaticAnalysis();
```
or even, as @phejl mentioned previously
```
        List<IResult> analysisResults = jfrEvents.automaticAnalysis();
```
where `jfrEvents.awaitEvents()` is called in `jfrEvents.automaticAnalysis()`

In the last 2 examples, `JfrAnalysis` would be an internal class.  In the first it would be user-facing.

What do you think is best in terms of user experience? To have the analysis to be exposed as the `JfrEvents` api, or as a separate, user-facing class that consumes `JfrEvents`?

Reply:
In terms of discoverability, I'd prefer the latter two, as you'll find the method easier than in comparison to it being static, declared on a class which you need to know about in the first place.